include/knowhere/index.h

```diff
@@ -116,6 +117,7 @@ class Index {
     Status
     Build(const DataSet& dataset, const Json& json) {
         Json json_(json);
+        normalize(dataset, json_);
```

Comment:
Let's put the COSINE check here in index.h, since if not COSINE we shouldn't even call a `normalize` function. Also, if the purpose is just to centralize the code, we can have a private inline function for it here.

Reply:
Also, use `IsMetricType` here, that is a good practice.

Reply:
updated